bluemix/configuration/core_config/bx_config_test.go

```diff
@@ -298,9 +301,72 @@ func TestVPCCRITokenURL(t *testing.T) {
+	newToken := "eyJraWQiOiIyMDE3MTAzMC0wMDowMDowMCIsImFsZyI6IkhTMjU2In0.eyJpYW1faWQiOiJpYW0tUHJvZmlsZS05NDQ5N2QwZC0yYWMzLTQxYmYtYTk5My1hNDlkMWIxNDYyN2MiLCJpZCI6IklCTWlkLXRlc3QiLCJyZWFsbWlkIjoiaWFtIiwianRpIjoiMDRkMjBiMjUtZWUyZC00MDBmLTg2MjMtOGNkODA3MGI1NDY4IiwiaWRlbnRpZmllciI6IlByb2ZpbGUtOTQ0OTdkMGQtMmFjMy00MWJmLWE5OTMtYTQ5ZDFiMTQ2MjdjIiwibmFtZSI6Ik15IFByb2ZpbGUiLCJzdWIiOiJQcm9maWxlLTk0NDk3ZDBkLTJhYzMtNDFiZi1hOTkzLWE0OWQxYjE0NjI3YyIsInN1Yl90eXBlIjoiUHJvZmlsZSIsImF1dGhuIjp7InN1YiI6ImNybjp2MTpzdGFnaW5nOnB1YmxpYzppYW0taWRlbnRpdHk6OmEvMThlMzAyMDc0OWNlNDc0NGIwYjQ3MjQ2NmQ2MWZkYjQ6OmNvbXB1dGVyZXNvdXJjZTpGYWtlLUNvbXB1dGUtUmVzb3VyY2UiLCJpYW1faWQiOiJjcm4tY3JuOnYxOnN0YWdpbmc6cHVibGljOmlhbS1pZGVudGl0eTo6YS8xOGUzMDIwNzQ5Y2U0NzQ0YjBiNDcyNDY2ZDYxZmRiNDo6Y29tcHV0ZXJlc291cmNlOkZha2UtQ29tcHV0ZS1SZXNvdXJjZSIsIm5hbWUiOiJteV9jb21wdXRlX3Jlc291cmNlIn0sImFjY291bnQiOnsiYm91bmRhcnkiOiJnbG9iYWwiLCJ2YWxpZCI6dHJ1ZSwiYnNzIjoiZmFrZV9ic3MifSwiaWF0IjoxNjI5OTI5NDYzLCJleHAiOjgwMjk5MzMwNjMsImlzcyI6Imh0dHBzOi8vaWFtLmNsb3VkLmlibS5jb20vaWRlbnRpdHkiLCJncmFudF90eXBlIjoidXJuOmlibTpwYXJhbXM6b2F1dGg6Z3JhbnQtdHlwZTpjci10b2tlbiIsInNjb3BlIjoiaWJtIG9wZW5pZCIsImNsaWVudF9pZCI6ImJ4In0.ACeIK_8Wi0QmgQ19w4J2OA0OKgC4zb6M6PuGuPTEY_E"
+	tests := []struct {
+		name       string
+		token      string
+		newToken   string
+		refresh    string
+		newRefresh string
+		isLoggedIn bool
+	}{
+		{
+			name:       "token is expired and refresh token is present",
+			token:      expiredToken,
+			newToken:   "bearer " + newToken, // on refresh the bearer header is append to the token
+			refresh:    oldToken,
+			newRefresh: newToken,
+			isLoggedIn: true,
+		},
+		{
+			name:       "token is expired and refresh token is NOT present",
+			token:      expiredToken,
+			newToken:   expiredToken,
+			refresh:    "",
+			newRefresh: "",
+			isLoggedIn: false,
+		},
+		{
+			name:       "token is not expired",
+			token:      newToken,
+			newToken:   newToken,
+			refresh:    newToken,
+			newRefresh: newToken,
+			isLoggedIn: true,
+		},
+	}
+
+	IAMEndpoints := models.Endpoints{}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(200)
+				fmt.Fprintf(w, "{\"access_token\": \"%s\", \"refresh_token\": \"%s\", \"token_type\": \"bearer\"}", newToken, newToken)
+			}))
+			defer ts.Close()
+
+			IAMEndpoints.PublicEndpoint = ts.URL
+			IAMEndpoints.PrivateEndpoint = ts.URL
+			IAMEndpoints.PrivateVPCEndpoint = ts.URL
+
+			config.SetIAMToken(test.token)
+			config.SetIAMRefreshToken(test.refresh)
```

Comment:
You can probably provide a mock refresh token that is more in line to what the server actually returns to better test your new functionality 🙂

Reply:
Will do